packages/babel-preset-env/scripts/build-data.js

```diff
@@ -58,19 +58,45 @@ const compatSources = ["es6", "es2016plus", "esnext"].reduce(
   (result, source) => {
     const data = require(`compat-table/data-${source}`);
     data.browsers = pickBy(envs, byTestSuite(source));
+    data.source = source;
     result.push(data);
     return result;
   },
   []
 );
 
-const interpolateAllResults = (rawBrowsers, tests) => {
-  const interpolateResults = res => {
+const interpolateAllResults = (rawBrowsers, tests, source) => {
+  const interpolateResults = (res, category) => {
     let browser;
     let prevBrowser;
     let result;
     let prevResult;
     let prevBid;
+    let ecmascript;
+
+    if (source === "es6") {
+      ecmascript = 2015;
+    } else if (source === "es2016plus") {
+      if (category.match(/2016/)) {
```

Comment:
Let's prevent `match` invocation more than once for 2017 and 2018 cases?
For example,
```
const esVersion = category.match(/201[6-8]/)
if (!esVersion) throw new Error('Could not find ecmascript...')
ecmascript = +esVersion[0];
```